### PR TITLE
SpinShare Update - Fixed Displaying Patreon User Span HTML

### PR DESCRIPTION
**Pull request by jy1263**
---
Name: SpinShare
About: Fixed presence showing innerHTML for users with patreon.
---
**Screenshots**
Before:
image

After:
image

> After your pull request has been made, you should have a screenshot and the code.

### CODE REVIEW

**Review by Bas950 — APPROVED** [review]

**Review by Alanexei — APPROVED** [review]

**Comment by Alanexei** [comment]
@jy1263 It seems that your presence was updated since you've made this pull request, or maybe you forgot to update your fork. You will need to resolve those conflicting files.

**Comment by jy1263** [comment]
thanks for the notice, yea i forgot to update my fork

**Review by Bas950 — APPROVED** [review]

File: websites/S/Spinsha.re/presence.ts [hunk 1]
```diff
@@ -4,67 +4,70 @@ const presence = new Presence({
...
+                presenceData.details = "Viewing Hot Charts";
+                break;
+            case "/popular":
+                presenceData.startTimestamp = browsingStamp;
+                presenceData.details = "Viewing Popular Charts";
+                break;
+            case "/legal":
+                presenceData.startTimestamp = browsingStamp;
+                presenceData.details = "Browsing Legal Information...";
+                presenceData.state = "Why?";
```
Comment (Alanexei) [inline websites/S/Spinsha.re/presence.ts:33]: I'd recommend removing this since this isn't very fitting.

File: websites/S/Spinsha.re/presence.ts [hunk 2]
```diff
@@ -4,67 +4,70 @@ const presence = new Presence({
...
+                break;
+            case "/support":
+                presenceData.startTimestamp = browsingStamp;
+                presenceData.details = "Browsing Support Information...";
+                presenceData.state = "Pls Donate Lol";
```
Comment (Alanexei) [inline websites/S/Spinsha.re/presence.ts:38]: I'd recommend removing this since this isn't very fitting.

File: websites/S/Spinsha.re/presence.ts [hunk 3]
```diff
@@ -4,67 +4,70 @@ const presence = new Presence({
...
+                break;
+            default:
+                //Idle
+                presenceData.startTimestamp = browsingStamp;
+                presenceData.details = "Idling";
+                presenceData.state = "Doing... Something?";
```
Comment (Alanexei) [inline websites/S/Spinsha.re/presence.ts:44]: I'd recommend removing this since this isn't very fitting.

**Review by Alanexei — CHANGES_REQUESTED** [review]

File: websites/S/Spinsha.re/presence.ts (hunk 3 above)
Comment (jy1263) [reply websites/S/Spinsha.re/presence.ts:44]: okay will do

File: websites/S/Spinsha.re/presence.ts [hunk 4]
```diff
@@ -4,67 +4,59 @@ const presence = new Presence({
...
+        }
+        else if (pathname.startsWith("/user")) {
+            presenceData.startTimestamp = browsingStamp;
+            presenceData.details = "Browsing User Profile:";
+            presenceData.state = (<HTMLElement> document.querySelector(".user-name")).innerText;
+        }
+        else if (pathname.startsWith("/search")) {
+            presenceData.startTimestamp = browsingStamp;
+            presenceData.details = "Searching";
+            presenceData.state = "🔍";
```
Comment (doomlerd) [inline websites/S/Spinsha.re/presence.ts:52]: Upload proof of this case.

File: websites/S/Spinsha.re/presence.ts [hunk 5]
```diff
@@ -4,67 +4,59 @@ const presence = new Presence({
...
+        }
+        else if (pathname.startsWith("/report")) {
+            presenceData.startTimestamp = browsingStamp;
+            presenceData.details = "Reporting Something...";
+            presenceData.state = "🔨";
```
Comment (doomlerd) [inline websites/S/Spinsha.re/presence.ts:57]: Upload proof of this case too.

**Review by doomlerd — CHANGES_REQUESTED** [review]

**Comment by jy1263** [comment]
Proof for Searching:
image

Proof for Reporting:
image

File: websites/S/Spinsha.re/presence.ts (hunk 5 above)
Comment (jy1263) [reply websites/S/Spinsha.re/presence.ts:57]: okie done

**Review by doomlerd — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**